src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -156,6 +160,38 @@ private CompletableFutures() {
     return collectingAndThen(toList(), CompletableFutures::allAsList);
   }
 
+  /**
+   * Apply an asynchronous operation to every entity in a stream, and collect all replies into a
+   * map.
+   *
+   * <p> Usage:
+   *
+   * <pre>{@code
+   *   collection.stream()
+   *      .collect(joinMap(this::someAsyncFunc))
+   *      .thenApply(this::consumeMap)
```

Comment:
IIUC, `consumeMap` is where the user of the library applies their business logic? It might be better to call it something like `useResultingMap`, `interpretResults`, or something like that. `consumeMap` sounds like it's a function in this library - I realise the same would apply to other javadoc comments.

Reply:
No reason to not do the right thing for new code at least. Updated to a hopefully more understandable
```java
collection.stream()
    .collect(joinMap(value -> asyncOperation(value)))
    .thenApply(map -> interpretResults(map))
```